### DISCUSSION TITLE
beforeRender lifecycle

### DISCUSSION THREAD

**Discussion by scripmanager** [livewire/livewire#1793] [1. Help]
Is the beforeRender() lifecycle discussed in your new DataTables screencasts available?

**Comment by nuernbergerA** [comment]
It's not released yet as v2.2.9 is the current version will be probably be released on Monday

**Reply by yabasha** [reply]
I am using "livewire/livewire" v2.5.5, but I am still getting `BadMethodCallException Method App\Http\Livewire\DataTable::beforeRender does not exist`

Please help?

**Reply by joshhanley** [reply]
@yabasha it was removed in this commit https://github.com/livewire/livewire/commit/de533eb000970e60449a09031f9b055ef2b59616

If you are using a trait, you can now use the rendering<TraitName> method. See at the end of this page https://laravel-livewire.com/docs/2.x/traits

Hope this helps!
